core/src/services/hdfs/backend.rs

```diff
@@ -262,12 +317,36 @@ impl Accessor for HdfsBackend {
             open_options.write(true);
         }
 
-        let f = open_options
-            .async_open(&p)
-            .await
-            .map_err(new_std_io_error)?;
+        if let Some(tmp_path) = &tmp_path {
+            let mut t = open_options
+                .async_open(tmp_path.to_str().unwrap())
+                .await
+                .map_err(new_std_io_error)?;
 
-        Ok((RpWrite::new(), HdfsWriter::new(f)))
+            let tmp_path = tmp_path.clone();
+            let target_path = target_path.clone();
+
+            t.flush().await.map_err(new_std_io_error)?;
```

Comment:
Please impelment those logic in `HdfsWriter` instead. We should rename file during `poll_close`.